File: src/RechercheService/models/produit.js
const mongoose = require("mongoose");
const { Schema, model } = mongoose;

const produitSchema = new Schema({
  _id: { type: String, required: true },
  reservePrice: Number,
  seller: String,
  winner: String,
  soldAmount: Number,
  currentHighBid: Number,
  createdAt: Date,
  updatedAt: Date,
  auctionStart: Date,
  auctionEnd: Date,
  status: String,
  make: String,
  name: String,
  year: Number,
  color: String,
  description: String,
  imageUrl: String,
  category: String,
  state: String,
});

produitSchema.index({ make: "text", name: "text", color: "text" });

const Produit = model("Produit", produitSchema);

module.exports = Produit;

File: src/EnchereService/Models/Produit.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace EnchereService.Models;

[Table("Items")]
public class Produit
{
    public Guid Id { get; set; }
    public string Make { get; set; }
    public string Name { get; set; }
    public int Year { get; set; }
    public string Color { get; set; }
    public string Description { get; set; }
    public string ImageUrl { get; set; }
    public string Category { get; set; }
    public Etat State { get; set; }
    public Enchere Auction { get; set; }
    public Guid AuctionId { get; set; }
}
